control_plane/src/endpoint.rs

```diff
@@ -774,7 +774,10 @@ impl Endpoint {
             spec.shard_stripe_size = stripe_size.map(|s| s.0 as usize);
         }
 
-        let client = reqwest::Client::new();
+        let client = reqwest::Client::builder()
+            .timeout(Duration::from_secs(30))
+            .build()
+            .unwrap();
```

Comment:
Why does this need to be in this PR? A timeout here could become a source of CI flakiness.

Reply:
Without time timeout, tests will hang forever if their flow of compute notifications the from controller don't happen properly.  That is a failure mode that comes  up if e.g. something goes wrong with registration.